pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -169,9 +172,52 @@ func AddToManager(mgr manager.Manager, r *ReconcileClusterPool, concurrentReconc
 		return err
 	}
 
+	// Watch for changes to ClusterDeploymentCustomizations
+	if err := c.Watch(
+		&source.Kind{Type: &hivev1.ClusterDeploymentCustomization{}},
+		handler.EnqueueRequestsFromMapFunc(
+			requestsForCDCResources(r.Client, r.logger)),
+	); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func requestsForCDCResources(c client.Client, logger log.FieldLogger) handler.MapFunc {
+	return func(o client.Object) []reconcile.Request {
+		cdc, ok := o.(*hivev1.ClusterDeploymentCustomization)
+		if !ok {
+			return nil
+		}
+
+		cpList := &hivev1.ClusterPoolList{}
+		if err := c.List(context.Background(), cpList, client.InNamespace(o.GetNamespace())); err != nil {
+			logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to list cluster pools for CDC resource")
+			return nil
+		}
+
+		var requests []reconcile.Request
+		for _, cpl := range cpList.Items {
+			if cpl.Spec.Inventory != nil {
```

Comment:
nit: I would reverse this and recover an indent
```suggestion
			if cpl.Spec.Inventory == nil {
				continue
			}
```